### PR TITLE
Write passthrough to meta data and add some missing fields

### PR DESCRIPTION
**Pull request by valentindavid**
Implement snap.yaml passthrough and add missing fields
`daemon_scope` and `activates_on`.

### CODE REVIEW

File: snapcraft/meta/snap_yaml.py [hunk 1]
```diff
@@ -281,6 +293,55 @@ def read(prime_dir: Path) -> SnapMetadata:
...
+        post_stop_command=app.post_stop_command,
+        start_timeout=app.start_timeout,
+        stop_timeout=app.stop_timeout,
+        watchdog_timeout=app.watchdog_timeout,
+        reload_command=app.reload_command,
+        restart_delay=app.restart_delay,
+        timer=app.timer,
+        daemon=app.daemon,
+        after=app.after if app.after else None,
+        before=app.before if app.before else None,
```
[the remark this answers is not in the document]
Comment (valentindavid) [reply snapcraft/meta/snap_yaml.py:324]: Right. I just moved the code. But I can change that.

File: snapcraft/meta/snap_yaml.py [hunk 2]
```diff
@@ -281,6 +293,55 @@ def read(prime_dir: Path) -> SnapMetadata:
     return SnapMetadata.unmarshal(data)
 
 
+def _write_app(app: App, assumes: Set[str]):
```
[the remark this answers is not in the document]
Comment (valentindavid) [reply snapcraft/meta/snap_yaml.py]: Renamed and added the return type annotation.

File: snapcraft/meta/snap_yaml.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (valentindavid) [reply snapcraft/meta/snap_yaml.py]: Done.

File: snapcraft/meta/snap_yaml.py [hunk 3]
```diff
@@ -365,6 +388,9 @@ def write(project: Project, prime_dir: Path, *, arch: str, arch_triplet: str):
         layout=project.layout,
         system_usernames=project.system_usernames,
     )
+    if project.passthrough:
```
[the remark this answers is not in the document]
Comment (valentindavid) [reply snapcraft/meta/snap_yaml.py:390]: Done.

File: snapcraft/meta/snap_yaml.py [hunk 4]
```diff
@@ -281,6 +293,55 @@ def read(prime_dir: Path) -> SnapMetadata:
...
+        stop_mode=app.stop_mode,
+        restart_condition=app.restart_condition,
+        install_mode=app.install_mode,
+        plugs=app.plugs,
+        aliases=app.aliases,
+        environment=app.environment,
+        command_chain=app.command_chain if app.command_chain else None,
+        sockets=app_sockets if app_sockets else None,
+        daemon_scope=app.daemon_scope if app.daemon_scope else None,
+        activates_on=app.activates_on if app.activates_on else None,
```
[the remark this answers is not in the document]
Comment (valentindavid) [reply snapcraft/meta/snap_yaml.py]: Done.

File: snapcraft/meta/snap_yaml.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (cmatsuoka) [reply snapcraft/meta/snap_yaml.py]: Ah, sorry, now I see this was in the original code. Thanks for refactoring that.

**Review by cmatsuoka — APPROVED** [review]
Thank you.

**Pull request merged**